### DISCUSSION TITLE
[Question] Why the trained results of the example of Imitation Learning in the Isaac Lab are very bad?

### DISCUSSION THREAD

**Discussion by tkdggg** [isaac-sim/IsaacLab#638] [Q&A]
### Question

I collected 120 demonstrations for training, but the trained results are still very poor. Sometimes, the robotic arm moves towards the cube, but it consistently fails to grasp the cube.

**Comment by Mayankm96** [comment]
Hard for me to say. We had done experiments in the past with imitation learning and it worked fine. Although our scenario was simple (lifting a cube with a random initial position and fixed target).

This depends a lot on your setup (observations, actions) and the model you're using.

**Reply by tkdggg** [reply]
Thank you for your reply! Could you tell me how many demonstrations you collected in your training?

**Reply by Oliverbansk** [reply]
Your code is really awesome! I'm running into the same problem. Could you tell me what setup (observation space / action space) , collection technique (keyboard or SpaceMouse) and demo collection strategy (going freely or following some protocols like moving along xyz-axis in a specific order)? Thank you!
